tests/labs/lie/functional/test_se3.py

```diff
@@ -30,7 +30,7 @@
 @pytest.mark.parametrize("batch_size", BATCH_SIZES_TO_TEST)
 @pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
 def test_op(op_name, batch_size, dtype):
-    import theseus.labs.lie_functional.se3 as se3
+    import theseus.labs.lie.functional.se3_impl as se3
```

Comment:
Any reason we renamed se3.py to se3_impl.py? Does `import theseus.labs.lie.functional.se3 as se3` cause issues?

Reply:
I think it's better if that file is hidden from users as much as possible, since it has a bunch of backend stuff that's not meant to be accessed. My idea is that the import that people will be using is `theseus.labs.lie.functional`, which provides the namespaces `se3` and `so3`. This `as se3` is just for the test, that's not meant to be the recommended usage.

Reply:
In fact, maybe we can should rename these packages as `theseus.labs.lie.functional._impl.so3`.

Reply:
Makes sense. I like the suggestion to add underscore to indicate 'internal'. Maybe `_se3_impl` is sufficient?

Reply:
Thinking here of how `torch` has a bunch of private [packages](https://github.com/pytorch/pytorch/tree/master/torch) in their repo.

Reply:
Ah. `_impl` as package sounds to generic. How about `_groups`?